pkg/metrics/collector_test.go

```diff
@@ -0,0 +1,376 @@
+package metrics
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	lxd "github.com/lxc/lxd/client"
+	lxdapi "github.com/lxc/lxd/shared/api"
+	mockclient "github.com/nieltg/lxd_exporter/test/mockgen/mock_client"
+	"github.com/nieltg/prom-example-testutil/pkg/testutil"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+)
+
+func prepare(t gomock.TestReporter) (
+	controller *gomock.Controller,
+	logger *log.Logger,
+	server *mockclient.MockInstanceServer,
+) {
+	controller = gomock.NewController(t)
+	logger = log.New(os.Stdout, "lxd_exporter: ", 0)
+	server = mockclient.NewMockInstanceServer(controller)
+	return
+}
+
+func collect(logger *log.Logger, server lxd.InstanceServer) {
+	ch := make(chan prometheus.Metric)
+	go func() {
+		for range ch {
+		}
+	}()
+
+	NewCollector(logger, server).Collect(ch)
+	close(ch)
+}
+
+func Example_collector_Collect_containerNamesError() {
```

Comment:
Can we change func name using standard go convention?

Reply:
This link refers to the standard Go convention: https://golang.org/pkg/testing/#hdr-Examples. The documentation doesn't state about the naming convention of private members, so I add underscore.

Adding underscore is needed for Go to recognize test functions. This issue has been acknowledged by @ekarisky from personal chat.